### PR TITLE
fix(WEB-6600): add error outline in CodeEditor when form is in error and fix windows build process

### PR DESCRIPTION
**Pull request by jkatofskyCoveo**
### Proposed Changes

- In the `CodeEditor`, now also display the error outline when there is an `error` coming from the form
- Fix some things along the way that were breaking when I tried to build the plasma repo on my Windows machine

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react-icons/bin/index.js [hunk 1]
```diff
@@ -87,5 +88,5 @@ rmSync(outDirPath, {recursive: true, force: true});
 rmSync('./dist', {recursive: true, force: true});
 fs.ensureDirSync(outDirPath);
 
-const svgs = globSync(`${iconsSourceDirPath}/**/*.svg`);
+const svgs = globSync(`${iconsSourceDirPath}/**/*.svg`).map((svgPath) => svgPath.split(path.sep).join(path.posix.sep));
```
Comment (toofff) [inline packages/react-icons/bin/index.js:91]: Why this change?

File: packages/components-props-analyzer/package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
         "compile": "node ../../scripts/build",
         "gen:props": "ts-node --project ./bin/tsconfig.json ./bin/index.ts",
         "lintfix": "../../node_modules/.bin/prettier --write \"**/*.{scss,ts,tsx,js,jsx,json,md,yml,html}\" && ../../node_modules/.bin/eslint -c ./.eslintrc.js \"src/**/*.{ts,tsx}\" --fix",
-        "prettify": "prettier --write 'src/components/*.ts'",
+        "prettify": "prettier --write \"src/components/*.ts\"",
```
Comment (toofff) [inline packages/components-props-analyzer/package.json:23]: For the tsx extension.

```suggestion
        "prettify": "prettier --write \"src/components/*.ts*\"",
```

File: packages/mantine/src/components/code-editor/CodeEditor.tsx [hunk 3]
```diff
@@ -143,6 +141,10 @@ export const CodeEditor: FunctionComponent<CodeEditorProps> = (props) => {
         }
     };
 
+    const [hasMonacoError, setHasMonacoError] = useState(false);
+    const renderErrorOutline = useMemo(() => !!error || hasMonacoError, [hasMonacoError, error]);
+    const {classes, theme} = useStyles({error: renderErrorOutline});
```
Comment (toofff) [inline packages/mantine/src/components/code-editor/CodeEditor.tsx:146]: Could you add some tests?

Comment (gdostie) [inline packages/mantine/src/components/code-editor/CodeEditor.tsx:146]: While at it we should probably make it overridable via the theme.

```suggestion
    const {classes, theme} = useStyles({error: renderErrorOutline}, {name: 'CodeEditor'});
```

File: packages/mantine/src/components/code-editor/CodeEditor.tsx [hunk 4]
```diff
@@ -143,6 +141,10 @@ export const CodeEditor: FunctionComponent<CodeEditorProps> = (props) => {
         }
     };
 
+    const [hasMonacoError, setHasMonacoError] = useState(false);
+    const renderErrorOutline = useMemo(() => !!error || hasMonacoError, [hasMonacoError, error]);
```
[the remark this answers is not in the document]
Comment (jkatofskyCoveo) [reply packages/mantine/src/components/code-editor/CodeEditor.tsx:145]: yeah fair point, force of habit to use the memo

File: packages/components-props-analyzer/package.json (hunk 2 above)
Comment (jkatofskyCoveo) [reply packages/components-props-analyzer/package.json:23]: Good idea

Comment (jkatofskyCoveo) [reply packages/components-props-analyzer/package.json:23]: Actually, it looks like it's only `ts` files in that folder...

File: packages/react-icons/bin/index.js (hunk 1 above)
Comment (jkatofskyCoveo) [reply packages/react-icons/bin/index.js:91]: `globSync` returns paths formatted with `\` instead of `/` when running the script on a windows machine, which breaks other parts of the script that the path strings to have `/`. This formats all the paths in the UNIX format.

File: packages/mantine/src/components/code-editor/CodeEditor.tsx (hunk 3 above)
Comment (jkatofskyCoveo) [reply packages/mantine/src/components/code-editor/CodeEditor.tsx]: added an assert in existing test to verify the border. It's really hard to test the case of a marker coming from `monaco` itself, since we mock the entire monaco editor.

**Review by toofff — APPROVED** [review]

**Review by gdostie — APPROVED** [review]
